mesa/model.py

```diff
@@ -77,25 +88,66 @@ def read_csv_city(self, filename: str):
         """
         return pd.read_csv(os.path.join(self.data_folder, filename + ".csv"))
 
+    def change_global_gender_ratio(self, target_ratio):
+        ratio_m = []
+        males_pop = 0
+        females_pop = 0
+        for i in self.areas:
+            details_list = self.population_details[i][i]
+            for detail in details_list:
+                if detail[2]:  # male
+                    males_pop += detail[5]
+                else:
+                    females_pop += detail[5]
+            ratio_m.append(males_pop / (males_pop + females_pop))
+        new_list = []
+        for i in self.areas:
+            details_list = self.population_details[i][i]
+            for detail in details_list:
+                if detail[2]:
+                    detail[5] = detail[5] * target_ratio / ratio_m[i - 1]
+                else:
+                    detail[5] = detail[5] * (1 - target_ratio) / ratio_m[i - 1]
+                new_list.append(detail)
+            self.population_details[i] = {i: new_list}
+
     def load_totals(self):
         local_db = self.read_csv_city("neighborhoods")
-        model.local = scen.from_db_to_dict("local", local_db)
+        self.local = scen.from_db_to_dict("local", local_db)
         population_db = self.read_csv_city("neukolln-totals")
-        model.population = scen.from_db_to_dict("population", population_db)
-        model.areas = list(model.population.keys())
+        self.population = scen.from_db_to_dict("population", population_db)
+        self.areas = list(self.population.keys())
         names = population_db["area_name"].values.tolist()
         population = population_db["sum(value)"].values.tolist()
         population_sum = sum(population)
         for i in range(len(model.areas)):
-            key = model.areas[i]
+            key = self.areas[i]
             self.area_population[key] = population[i] / population_sum * self.total_citizens
             self.area_names[key] = names[i]
+        for a in self.areas:
+            details = []
+            for index, row in self.population_details_db.iterrows():
+                if row['area_code'] == a:
+                    details.append(row.values)
+            self.population_details[a] = {a: details}
+
+    def setup_world(self):
```

Comment:
Same here: No need for a function, just create a variable. Unless there are calculations to be done in the future.

Reply:
proton-T prevede la possibilità di modificare i dati demografici in corso d'opera. Io lascerei così e cambierei solo se questa possibilità non si verfica